.github/workflows/ci.yml

```diff
@@ -17,26 +17,42 @@ env:
 jobs:
   test:
     runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: ["3.7"] # Cairo was tested with this version
+        node-version: ["16"]
     steps:
       - uses: actions/checkout@v3
 
       - name: Use Node.js
         uses: actions/setup-node@v3
         with:
-          node-version: "16"
+          node-version: ${{ matrix.node-version }}
           cache: "yarn"
 
       - name: Setup Python
         uses: actions/setup-python@v3
         with:
-          python-version: 3.7 # Cairo was tested with this version
+          python-version: ${{ matrix.python-version }}
+
+      - name: Cache Python environment
+        uses: actions/cache@v3
+        with:
+          path: .venv
+          key: venv-${{ runner.os }}-py${{ matrix.python-version }}-${{ hashFiles('poetry.lock') }}
 
       - name: Install dependencies
         run: |
           python -m venv .venv
           source .venv/bin/activate
           make
 
+      - name: Static analysis for Solidity
+        run: ./scripts/static_analysis_solidity.sh
+
+      - name: Static analysis for Cairo
+        run: python scripts/static_analysis_cairo.py
```

Comment:
We should fail the ci if the sarif produced by the ci is different to the sarif checked into the repo

Reply:
To ensure that we commit vulnerabilities files in each PR? I'm not sure yet it's a good idea to do so

Reply:
We could add these in a pre-push hook, but these analyses take too much time to be run for each commit